### PR TITLE
Add CLI::table

### PR DESCRIPTION
**Pull request by natanfelles**
Working on it...

### CODE REVIEW

File: system/CLI/CLI.php [hunk 1]
```diff
@@ -750,6 +750,121 @@ public static function getOptionString(): string
...
+
+			// Set the thead and table borders-bottom
+			if ($i === 0 && ! empty($thead) || $i + 1 === $total_rows)
+			{
+				$table .= $cols . PHP_EOL;
+			}
+		}
+
+		//fwrite(STDOUT, $table . PHP_EOL);
+		return $table;
```
Comment (natanfelles) [inline system/CLI/CLI.php]: The intention here is to be able to show the output and not return but when using `fwrite()` I am not being able to make the test pass ...

I tried with `ob_start()` and nothing... Even no success with `CLITestStreamFilter::$buffer`...

Comment (natanfelles) [reply system/CLI/CLI.php]: Saned.

**Comment by natanfelles** [comment]
This enable us to create tables like:

```php
$users = new Users;
$users = $users->select(['id', 'name', 'email', 'country'])->findAll(10);
CLI::table($users, ['ID', 'Name', 'Email', 'Country']);
```

```
+----+------------------+-----------------------+---------+
| ID | Name             | Email                 | Country |
+----+------------------+-----------------------+---------+
| 1  | Derek Jones      | derek@world.com       | US      |
| 2  | Ahmadinejad      | ahmadinejad@world.com | Iran    |
| 3  | Richard A Causey | richard@world.com     | US      |
| 4  | Chris Martin     | chris@world.com       | UK      |
+----+------------------+-----------------------+---------+
```

and...

```php
$router = \Config\Services::routes();

$methods = [
	'*',
	'options',
	'get',
	'head',
	'post',
	'put',
	'delete',
	'trace',
	'connect',
	'cli',
];

foreach ($methods as $method)
{
	foreach ($router->getRoutes($method) as $from => $to)
	{
		$routes[] = [
			$from,
			$method,
			$to,
		];
	}
}
array_multisort($routes);

CLI::table($routes ?? [], ['From', 'Method', 'To']);
```

```
+----------------------------+---------+------------------------------------------------+
| From                       | Method  | To                                             |
+----------------------------+---------+------------------------------------------------+
| /                          | *       | \App\Controllers\Home::index                   |
| /                          | cli     | \App\Controllers\Home::index                   |
| /                          | connect | \App\Controllers\Home::index                   |
| /                          | delete  | \App\Controllers\Home::index                   |
| /                          | get     | \App\Controllers\Home::index                   |
| /                          | head    | \App\Controllers\Home::index                   |
| /                          | options | \App\Controllers\Home::index                   |
| /                          | post    | \App\Controllers\Home::index                   |
| /                          | put     | \App\Controllers\Home::index                   |
| /                          | trace   | \App\Controllers\Home::index                   |
| ci(.*)                     | cli     | \CodeIgniter\CLI\CommandRunner::index/$1       |
| migrations                 | cli     | \CodeIgniter\Commands\MigrationsCommand::index |
| migrations/([^/]+)         | cli     | \CodeIgniter\Commands\MigrationsCommand::$1    |
| migrations/([^/]+)/([^/]+) | cli     | \CodeIgniter\Commands\MigrationsCommand::$1/$2 |
+----------------------------+---------+------------------------------------------------+
```

**Comment by natanfelles** [comment]
I get it. But... it seems a bit complicated for now...

This output have 90 columns:

```
+----------------------------+---------+------------------------------------------------+
| From                       | Method  | To                                             |
+----------------------------+---------+------------------------------------------------+
| migrations                 | cli     | \CodeIgniter\Commands\MigrationsCommand::index |
| migrations/([^/]+)         | cli     | \CodeIgniter\Commands\MigrationsCommand::$1    |
| migrations/([^/]+)/([^/]+) | cli     | \CodeIgniter\Commands\MigrationsCommand::$1/$2 |
+----------------------------+---------+------------------------------------------------+
```

Supposing that the maximum width is 80. So should the output be similar to that?

```
+----------------------------+---------+--------------------------------------+
| From                       | Method  | To                                   |
+----------------------------+---------+--------------------------------------+
| migrations                 | cli     | \CodeIgniter\Commands\MigrationsComm |
|                            |         | and::index                           |
| migrations/([^/]+)         | cli     | \CodeIgniter\Commands\MigrationsComm |
|                            |         | and::$1                              |
| migrations/([^/]+)/([^/]+) | cli     | \CodeIgniter\Commands\MigrationsComm |
|                            |         | and::$1/$2                           |
+----------------------------+---------+--------------------------------------+
```

What seems to me more complicated is, and if the screen is only 20, or 10 columns?

I can try to do it, although I find it more practical to increase or decrease the font size on the terminal screen. But, of course, a very small font can not be read ...

LXTerminal, Gnome-Terminal, Tilix and Terminator allow to do this with `Ctrl + +`, `Ctrl + -` or `Ctrl + mouse wheel`.

**Comment by lonnieezell** [comment]
You know - you're probably right. Trying to do that correctly would mean juggling the sizes of however many columns exist, and trying to resize each proportionately... sounds like a giant pain. So, ignore me for now :) I'll approve this one. 

Thanks for adding this feature, by the way!

**Pull request merged**